internal/graph/check.go

```diff
@@ -790,6 +792,28 @@ func buildTupleKeyConditionFilter(ctx context.Context, reqCtx *structpb.Struct,
 	}
 }
 
+type usersetDetailsFunc func(*openfgav1.TupleKey) (string, string, error)
+
+// buildUsersetDetails given tuple doc:1#viewer@group:2#member will return group#member, 2, nil.
+// This util takes into account pre-computed relationships, otherwise it will resolve it from the target UserType.
+func buildUsersetDetails(typesys *typesystem.TypeSystem, computedRelation, userType string) usersetDetailsFunc {
+	return func(t *openfgav1.TupleKey) (string, string, error) {
+		cr := computedRelation
+		object, relation := tuple.SplitObjectRelation(t.GetUser())
+		objectType, objectID := tuple.SplitObject(object)
+		if cr == "" {
+			terminalRelations := typesys.GetTerminalRelations(objectType, relation, userType)
```

Comment:
Looks like we have seen the following pattern a few times.  Maybe we should create a helper method in typesys so that the error message is standardized
```
x := typesys.GetTerminalRelations(a, b, c)
if len(x) != 1 {
  return fmt.Errorf("Foo")
}
y := x[0]
```